src/lib/operators/sort.cpp

```diff
@@ -139,6 +140,7 @@ class Sort::SortImplMaterializeOutput {
         if (chunk_offset_out > 0u) {
           auto value_segment = std::make_shared<ValueSegment<ColumnDataType>>(std::move(value_segment_value_vector),
                                                                               std::move(value_segment_null_vector));
+          value_segment->set_sort_order(order_by_mode);
```

Comment:
Has nothing to do with your change. You changes just shows that lines 129 - 132 are duplicated.